spec/features/merge_pact_spec.rb

```diff
@@ -0,0 +1,57 @@
+describe "Merging a pact" do
+
+  let(:pact_content) { load_fixture('a_consumer-a_provider.json') }
+  let(:path) { "/pacts/provider/A%20Provider/consumer/A%20Consumer/version/1.2.3" }
+  let(:response_body_json) { JSON.parse(subject.body) }
+
+  subject do
+    patch path, pact_content, {'CONTENT_TYPE' => 'application/json' }; last_response
+  end
+
+  context "when a pact for this consumer version does not exist" do
+    it "returns a 201 Created" do
+      expect(subject.status).to be 201
+    end
+
```

Comment:
Actually, the patch RFC states that it should create a new resource if the
resource does not exist. It has all the same semantics as a put.

On Fri, 29 Jul 2016, 08:17 Beth Skurrie <notifications@github.com> wrote:

> In spec/features/merge_pact_spec.rb
> <https://github.com/bethesque/pact_broker/pull/52#discussion_r72713344>:
>
> > @@ -0,0 +1,57 @@
> > +describe "Merging a pact" do
> > +
> > +  let(:pact_content) { load_fixture('a_consumer-a_provider.json') }
> > +  let(:path) { "/pacts/provider/A%20Provider/consumer/A%20Consumer/version/1.2.3" }
> > +  let(:response_body_json) { JSON.parse(subject.body) }
> > +
> > +  subject do
> > +    patch path, pact_content, {'CONTENT_TYPE' => 'application/json' }; last_response
> > +  end
> > +
> > +  context "when a pact for this consumer version does not exist" do
> > +    it "returns a 201 Created" do
> > +      expect(subject.status).to be 201
> > +    end
> > +
>
> I'm not keen on having PATCH create the resource when it doesn't exist,
> but I can see why you did it.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bethesque/pact_broker/pull/52/files/03c03b4cf39af8fd3da0c2406292dc3a2d398dc4#r72713344>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA0tsrbNDErQRndfa76rdnsqAqMizxNgks5qaSpggaJpZM4JXnPu>
> .
>